int32 goal
---
string message